File: client/archive/android/app/src/main/java/com/solutioncrafts/BriskVTU/feature/beneficiaries/BeneficiaryApi.kt
package com.solutioncrafts.BriskVTU.feature.beneficiaries

import kotlinx.serialization.Serializable
import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.PATCH
import retrofit2.http.POST
import retrofit2.http.Path

@Serializable
data class BeneficiaryResponse(
    val id: String,
    val userId: String,
    val name: String,
    val serviceType: String,
    val providerServiceId: String,
    val category: String,
    val isFavorite: Boolean,
    val metadata: Map<String, String>? = null,
    val lastUsedAt: String,
    val createdAt: String
)

@Serializable
data class CreateBeneficiaryRequest(
    val name: String,
    val serviceType: String,
    val providerServiceId: String,
    val category: String,
    val metadata: Map<String, String>? = null
)

@Serializable
data class UpdateBeneficiaryRequest(
    val name: String? = null,
    val isFavorite: Boolean? = null
)

interface BeneficiaryApi {
    @GET("beneficiaries")
    suspend fun listBeneficiaries(): List<BeneficiaryResponse>

    @POST("beneficiaries")
    suspend fun createBeneficiary(@Body request: CreateBeneficiaryRequest): BeneficiaryResponse

    @PATCH("beneficiaries/{id}")
    suspend fun updateBeneficiary(
        @Path("id") id: String,
        @Body request: UpdateBeneficiaryRequest
    ): BeneficiaryResponse

    @DELETE("beneficiaries/{id}")
    suspend fun deleteBeneficiary(@Path("id") id: String)
}

File: client/archive/ios/BriskVTU/Features/Beneficiaries/BeneficiaryAPI.swift
import Foundation

struct BeneficiaryResponse: Codable, Identifiable {
    let id: String
    let userId: String
    let name: String
    let serviceType: String
    let providerServiceId: String
    let category: String
    let isFavorite: Bool
    let metadata: [String: String]?
    let lastUsedAt: String
    let createdAt: String
}

struct CreateBeneficiaryRequest: Codable {
    let name: String
    let serviceType: String
    let providerServiceId: String
    let category: String
    let metadata: [String: String]?
}

struct UpdateBeneficiaryRequest: Codable {
    let name: String?
    let isFavorite: Bool?
}


final class BeneficiaryAPI {
    static let shared = BeneficiaryAPI()
    private let client = APIClient.shared
    
    func listBeneficiaries() async throws -> [BeneficiaryResponse] {
        let res: BaseResponse<[BeneficiaryResponse]> = try await client.request(endpoint: "/beneficiaries", method: "GET")
        return res.data ?? []
    }
    
    func createBeneficiary(request: CreateBeneficiaryRequest) async throws -> BeneficiaryResponse {
        let data = try JSONEncoder().encode(request)
        let res: BaseResponse<BeneficiaryResponse> = try await client.request(endpoint: "/beneficiaries", method: "POST", body: data)
        guard let beneficiary = res.data else { throw APIError.noData }
        return beneficiary
    }
    
    func updateBeneficiary(id: String, request: UpdateBeneficiaryRequest) async throws -> BeneficiaryResponse {
        let data = try JSONEncoder().encode(request)
        let res: BaseResponse<BeneficiaryResponse> = try await client.request(endpoint: "/beneficiaries/\(id)", method: "PATCH", body: data)
        guard let beneficiary = res.data else { throw APIError.noData }
        return beneficiary
    }
    
    func deleteBeneficiary(id: String) async throws {
        let _: BaseResponse<EmptyResponse> = try await client.request(endpoint: "/beneficiaries/\(id)", method: "DELETE")
    }
}
